data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
-                if len(train_smiles.intersection(test_split_smiles)) > 0:
-                    raise ValueError("Split failed, train and test overlap")
+def get_yaml_files(data_dir: Union[str, Path]) -> List[str]:
+    """Returns all yaml files in the data_dir directory."""
+    return glob(os.path.join(data_dir, "**", "*.yaml"), recursive=True)
 
 
-TRACKED_DATASETS = []
+def run_transform(file):
+    if not os.path.exists(os.path.join(os.path.dirname(file), "data_clean.csv")):
+        subprocess.run(
+            ["python", "transform.py"],
+            cwd=os.path.dirname(file),
+        )
 
 
-def per_repr(
-    repr_col: str,
+def remaining_split(
+    data_dir,
+    override: bool = False,
     seed: int = 42,
-    train_size: float = 0.8,
-    val_size: float = 0.0,
-    test_size: float = 0.2,
-    path: str = "*/data_clean.csv",
+    debug: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+):
+    # make deterministic
+    np.random.seed(seed)
+    random.seed(seed)
+    dask_random_state = da.random.RandomState(seed)
+
+    yaml_files = get_yaml_files(data_dir)
+    non_smiles_yaml_files = [
+        file for file in yaml_files if not has_yaml_file_smiles_column(file)
+    ]
+
+    # if we debug, we only run split on the first 5 datasets
+    if debug:
+        non_smiles_yaml_files = non_smiles_yaml_files[:5]
+
+    def assign_split(train_frac, val_frac, test_frac, random_state):
+        # Assign random number for train/remaining split using the random state
+        random_value = random_state.random_sample()
+        if random_value < train_frac:
+            return "train"
+        elif random_value < train_frac + test_frac:
+            return "test"
+        else:
+            return "valid"
+
+    # we run random splitting for each dataset
+    for file in non_smiles_yaml_files:
+        print(f"Processing {file}")
+        run_transform(file)
+
+        ddf = dask.dataframe.read_csv(
+            os.path.join(os.path.dirname(file), "data_clean.csv")
+        )
+        meta = ("split", "object")  # Meta defines the structure of the new column
+        ddf["split"] = ddf.apply(
+            assign_split,
+            axis=1,
+            meta=meta,
+            args=(train_frac, val_frac, test_frac, dask_random_state),
+        )
+        split_counts = ddf["split"].value_counts().compute()
+
+        print(f"Dataset {file} has {len(ddf)} datapoints. Split sizes: {split_counts}")
+
+        # we then write the new data_clean.csv file
+        # if the override option is true, we write this new file to `data_clean.csv` in the same directory
+        # otherwise, we copy the old `data_clean.csv` to `data_clean_old.csv`
+        # and write the new file to `data_clean.csv`
+        if not override:
+            os.rename(
+                os.path.join(os.path.dirname(file), "data_clean.csv"),
+                os.path.join(os.path.dirname(file), "data_clean_old.csv"),
```

Comment:
maybe add another arg to not keep the old file, as I guess we will run into storage issues with that